actionpack/lib/action_dispatch/request/utils.rb

```diff
@@ -73,6 +77,28 @@ def self.handle_array(params)
           list
         end
       end
+
+      class BinaryParamEncoder < ParamEncoder # :nodoc:
```

Comment:
I think we don't need to inherit from `ParamEncoder` since we are not using any of the shared code.